pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,39 @@
+package ipsec
+
+import (
+	"strconv"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedState     string
+	recordedLegacyAPI string
+)
+
+func init() {
+	ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+		Name: "cluster_network_ipsec_state",
+		Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+	}, []string{"mode", "is_legacy_api"})
+	legacyregistry.MustRegister(ipsecStateGauge)
+	recordedState = ""
+	recordedLegacyAPI = ""
+}
+
+func UpdateIPsecTelemetry(state string, isLegacyAPI bool) {
+	klog.V(5).Infof("IPsec mode(prev): %s(%s), isLegacyAPI(prev): %v(%s)",
+		state, recordedState, isLegacyAPI, recordedLegacyAPI)
+	isLegacyAPIStr := strconv.FormatBool(isLegacyAPI)
```

Comment:
i think you need another state besides true and false - if ipsec is disabled, then the third state should be `unknown` otherwise it looks strange defining is it legacy api or not when its not even configured. WDYT?